### ISSUE TITLE
Feature Request: API for Custom Animations

### ISSUE THREAD

**Issue by Jipok** [yshui/compton#1377]
Current picom animations are great, but they're limited by DSL capabilities. Would it be possible to add API/IPC support or scripting (e.g., Lua) for custom animations?

Use case: I'd like to trigger animations directly from AwesomeWM config to create effects similar to niri's workspace switching (https://github.com/YaLTeR/niri?tab=readme-ov-file#video-demo). With current picom capabilities, implementing such animations seems impossible without hacky workarounds like dynamic window class changes.

I believe many users have interesting animation ideas, but the DSL limitations make them technically unfeasible, even though they shouldn't be complex to implement with proper API/scripting support.

**Comment by kovasap** [comment]
I am interested in something like this for workspace switching as well.  Did you ever find a workaround/solution that does not require the addition of this API?

**Comment by Jipok** [comment]
No. I doubt there is anything that works. But to be fair, I wasn't looking for anything. I can assume that adding an API is the easiest way.
